src/users/repositories/user-items.repository.ts

```diff
@@ -0,0 +1,81 @@
+import { Injectable } from '@nestjs/common';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { UserItem } from '../entities/user-item.entity';
+import { Item } from 'src/items/entities/item.entity';
+
+interface WhereClause {
+  userId: number;
+  mainCategoryId?: number;
+  subCategoryId?: number;
+  isEquipped?: boolean;
+}
+
+@Injectable()
+export class UserItemsRepository {
+  constructor(private readonly prisma: PrismaService) {}
+
+  //아이템 목록 조회
+  async findUserItems(
+    page: number,
+    limit: number,
+    where: WhereClause,
+  ): Promise<(UserItem & { item: Item })[]> {
+    return this.prisma.userItem.findMany({
+      where: {
+        userId: where.userId,
+        ...(where.mainCategoryId && { mainCategoryId: where.mainCategoryId }),
+        ...(where.subCategoryId && { subCategoryId: where.subCategoryId }),
+      },
+      skip: (page - 1) * limit,
+      take: limit,
+      include: {
+        item: true,
+      },
+    });
+  }
```

Comment:
```suggestion
  async findUserItems(
    page: number,
    limit: number,
    where: WhereClause,
  ): Promise<(UserItem & { item: Item })[]> {
    return this.prisma.userItem.findMany({
      where: {
        userId: where.userId,
         Item:{
             ...(where.mainCategoryId && { mainCategoryId: where.mainCategoryId }),
             ...(where.subCategoryId && { subCategoryId: where.subCategoryId }),
        }
      },
      skip: (page - 1) * limit,
      take: limit,
      include: {
        item: true,
      },
    });
  }
```
userItems에서 메인카테고리와 서브카테고리 조회를 위해 테이블에 id 속성을 새로 만들었는데,
prisma 클라이언트 쿼리문에서 where 안에  Item을 가지고 조회를 하면 id속성 추가 없이 가능할 거 같습니다.